src/content_mapblock.cpp

```diff
@@ -800,8 +802,34 @@ void MapblockMeshGenerator::drawGlasslikeFramedNode()
 	for (int face = 0; face < 6; face++) {
 		if (nb[face])
 			continue;
+
 		tile = glass_tiles[face];
-		drawAutoLightedCuboid(glass_faces[face]);
+		// Face at Z-
+		v3f vertices[4] = {
+			v3f(-BS / 2,  BS / 2, -BS / 2),
+			v3f( BS / 2,  BS / 2, -BS / 2),
+			v3f( BS / 2, -BS / 2, -BS / 2),
+			v3f(-BS / 2, -BS / 2, -BS / 2),
+		};
+
+		for (v3f &vertex : vertices) {
+			switch (face) {
+				case D6D_ZP:
+					vertex.rotateXZBy(180); break;
+				case D6D_YP:
+					vertex.rotateYZBy( 90); break;
+				case D6D_XP:
+					vertex.rotateXZBy( 90); break;
+				case D6D_ZN:
+					vertex.rotateXZBy(  0); break;
```

Comment:
Unnecessary transformation.
```suggestion
					break;
```

Reply:
This was copied from 'glasslike', but yes will consider.